parsec/backend/postgresql/user_queries/find.py

```diff
@@ -97,45 +105,44 @@ async def query_find_humans(
     omit_revoked: bool,
     omit_non_human: bool,
 ) -> Tuple[List[HumanFindResultItem], int]:
+    if page >= 1:
+        offset = (page - 1) * per_page
+    else:
+        return ([], 0)
+
+    q = _q_human_factory(
+        query=True if query else False,
+        omit_revoked=True if omit_revoked else False,
+        omit_non_human=omit_non_human,
+        offset=offset,
+        limit=per_page,
+    )
     if query:
-
-        if omit_revoked:
-            q = _q_human_factory(query=True, omit_revoked=True, omit_non_human=omit_non_human)
-            args = q(organization_id=organization_id, now=pendulum_now(), query=query)
-
-        else:
-            q = _q_human_factory(query=True, omit_revoked=False, omit_non_human=omit_non_human)
-            args = q(organization_id=organization_id, now=pendulum_now(), query=query)
-
+        args = q(organization_id=organization_id, now=pendulum_now(), query=query)
     else:
-
-        if omit_revoked:
-            q = _q_human_factory(query=False, omit_revoked=True, omit_non_human=omit_non_human)
-            args = q(organization_id=organization_id, now=pendulum_now())
-
-        else:
-            q = _q_human_factory(query=False, omit_revoked=False, omit_non_human=omit_non_human)
-            args = q(organization_id=organization_id, now=pendulum_now())
+        args = q(organization_id=organization_id, now=pendulum_now())
 
     raw_results = await conn.fetch(*args)
-
+    if raw_results:
+        total = raw_results[0][4]
+    else:
+        total = 0
     humans = [
         HumanFindResultItem(
             user_id=UserID(user_id),
             human_handle=HumanHandle(email=email, label=label),
             revoked=revoked,
         )
-        for user_id, email, label, revoked in raw_results
+        for user_id, email, label, revoked, _ in raw_results
         if email is not None
     ]
     non_humans = [
         HumanFindResultItem(user_id=UserID(user_id), human_handle=None, revoked=revoked)
-        for user_id, email, label, revoked in raw_results
+        for user_id, email, label, revoked, _ in raw_results
         if email is None
     ]
     results = [
-        *sorted(humans, key=lambda x: (x.human_handle.label.lower(), x.user_id.lower())),
+        *sorted(humans, key=lambda r: r.human_handle.label),
```

Comment:
and no more need for sorting outside of postgresql also !

Reply:
Done